### PR TITLE
Closes #72 - Rework CTI documentation

### LINKED ISSUE #72: CTI documentations outdated
The user assignment parts were added after documentary for CTI have been created.
This leads to an outdated documentation

### PR DESCRIPTION
**Pull request by MrGeneration**
This PR updates all three CTI integration parts to reflect the current situation within Zammad.

### CODE REVIEW

File: system/integrations/cti/placetel.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+.. include:: /system/integrations/cti/includes/introduction.include.rst
+
+Requirements
+------------
+
+Please provide the following requirements:
+
+   * You need an administrative placetel account for your organization
+   * Your Zammad instance must be allowed to communicate to external services
+   * Placetel must be able to reach your Zammad instance
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/placetel.rst:18]: Dot at the end of the sentences.

File: system/integrations/cti/placetel.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+   provider-configurations/placetel
+
+.. include:: /system/integrations/cti/includes/introduction.include.rst
+
+Requirements
+------------
+
+Please provide the following requirements:
+
+   * You need an administrative placetel account for your organization
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/placetel.rst:16]: "Placetel", not "placetel".

File: system/integrations/cti/provider-configurations/placetel.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
+Setup placetel connection for Zammad
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/provider-configurations/placetel.rst:1]: "Placetel", not "placetel".

File: system/integrations/cti/placetel.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+   * Your Zammad instance must be allowed to communicate to external services
+   * Placetel must be able to reach your Zammad instance
+
+:doc:`provider-configurations/placetel`
+   Learn how to configure placetel to enable Zammad and placetel to communicate
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/placetel.rst:21]: "Placetel", not "placetel".

File: system/integrations/cti/placetel.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+.. include:: /system/integrations/cti/includes/important-dont-forget-saving.include.rst
+
+Endpoint Settings
+   The here listed endpoint settings are relevant for the
+   :doc:`integration configuration with placetel <provider-configurations/placetel>`.
+
+   Endpoint
+      This endpoint will be required for the Zammad integration within the
+      placetel web interface.
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/placetel.rst:42]: "Placetel", not "placetel".

File: system/integrations/cti/provider-configurations/placetel.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+====================================
+
+.. note::
+
+   This configuration step requires a *full* administrative placetel account.
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/provider-configurations/placetel.rst:6]: "Placetel", not "placetel".

File: system/integrations/cti/provider-configurations/placetel.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+         Apply your changes by using the "Save" button on the bottom of the
+         placetel integration page and activate the placetel integration.
+
+   .. figure:: /images/system/integrations/cti/placetel/step2-create-api-token-for-zammad.gif
+      :alt: Screencast showing how to retrieve an API token from placetel for Zammad
+
+Step 3: Restrict the numbers to notify on
+   Having a lot of numbers that shouldn't be used for notifying Zammad?
+   Within the the Integrations page of the placetel web interface, go to
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/provider-configurations/placetel.rst:56]: "Placetel", not "placetel".

File: system/integrations/cti/provider-configurations/placetel.rst [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+   "Notify API".
+
+   Lower on the page placetel allows you to restrict the numbers to notify on.
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/provider-configurations/placetel.rst:59]: "Placetel", not "placetel".

File: system/integrations/cti/provider-configurations/placetel.rst [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+   .. figure:: /images/system/integrations/cti/placetel/external-routing-api.png
+      :alt: Screenshot showing a sample selection of phone numbers to use for the placetels notify API
+
+   .. hint::
+
+      This menu point also provides a API request log from placetels view.
+      Just open "Recent responses of your API endpoint" to learn more.
+
+      If you want to see Zammads perspective, use the "Recent Logs" part from
+      within the placetel integration page.
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/provider-configurations/placetel.rst:71]: "Placetel", not "placetel".

File: system/integrations/cti/provider-configurations/placetel.rst [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+      .. figure:: /images/system/integrations/cti/placetel/recent-responses-of-your-api-endpoint.png
+         :alt: Screenshot showing sample log entries for placetels API calls to Zammad
+
+Step 4 (optional): Further configurations for placetel
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/provider-configurations/placetel.rst:76]: "Placetel", not "placetel".

File: system/integrations/cti/sipgate.rst [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+------------
+
+Please provide the following requirements:
+
+   * You need an administrative sipgate account for your organization
+   * Zammad requires a ``sipgate.io`` addon from the feature store
+      * Please note that API calls are not free with sipgate.
+        Rates may apply and differ from account to account.
+   * Your Zammad instance must be allowed to communicate to external services
+   * Sipgate must be able to reach your Zammad instance
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/sipgate.rst:21]: Dot at the end of the sentences.

File: system/integrations/cti/sipgate.rst [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+:doc:`provider-configurations/sipgate`
+   Learn how to configure sipgate to enable Zammad and sipgate to communicate
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/sipgate.rst:24]: "Sipgate", not "sipgate".

File: system/integrations/cti/includes/recent-logs.include.rst [hunk 13]
```diff
@@ -0,0 +1,27 @@
+Recent Logs
```
Comment (YetAnotherGerrit) [inline system/integrations/cti/includes/recent-logs.include.rst:1]: "logs", not "Logs".

**Review by YetAnotherGerrit — CHANGES_REQUESTED** [review]
A lot of the markings are similar, a lot of proper nouns/names that need to be capitalized. I may have missed some: Placetel/Sipgate.

File: system/integrations/cti/includes/recent-logs.include.rst (hunk 13 above)
Comment (MrGeneration) [reply system/integrations/cti/includes/recent-logs.include.rst:1]: This heading actually reflects the heading within the Zammad UI.

image


Do you think that this is a bad idea?

File: system/integrations/cti/generic.rst [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+Requirements
+------------
+
+Please provide the following requirements:
+
+   * a telephone system that supports webhooks (outbound)
+     (best option are those that allow customizing the calls)
+   * a unique Call-ID during the call session
+   * call event data (from and to numbers, direction)
```
[the remark this answers is not in the document]
Comment (MrGeneration) [reply system/integrations/cti/generic.rst]: This indirectly has context to the generic CTI API section and means that the PBX has to provide at least the following information:

* call direction (in / out)
* a from caller ID ("who calls?")
* a to caller ID ("who is called?")

This is in direct relevance to the API documentation:
https://docs.zammad.org/en/latest/cti/api-intro.html

This is why it's also pointed out below the bullet points:
image

File: system/integrations/cti/includes/recent-logs.include.rst (hunk 13 above)
Comment (YetAnotherGerrit) [reply system/integrations/cti/includes/recent-logs.include.rst:1]: From my understanding it is "Recent logs", not "Recent Logs", as it is not a proper name. But to provide consistency with the Zammad UI, I support your suggestion to keep it as "Recent Logs".

**Review by YetAnotherGerrit — APPROVED** [review]

**Pull request merged**